src/solve/execute-route.ts

```diff
@@ -0,0 +1,308 @@
+      throw new Error("No fee info found");
+    }
+
+    let gasNeeded = 200000;
+    if (
+      route.does_swap &&
+      route.swap_venue?.chain_id === multiHopMsg.chain_id
+    ) {
+      gasNeeded = 1000000;
+    }
+
+    let averageGasPrice = 0;
+    if (feeInfo.average_gas_price) {
+      averageGasPrice = feeInfo.average_gas_price;
+    }
+
+    const amountNeeded = averageGasPrice * gasNeeded;
+
+    const balance = await client.getBalance(
+      userAddresses[multiHopMsg.chain_id],
+      feeInfo.denom
+    );
+
+    if (parseInt(balance.amount) < amountNeeded) {
+      throw new Error(
+        `Insufficient fee token to initiate transfer on ${multiHopMsg.chain_id}. Need ${amountNeeded} ${feeInfo.denom}, but only have ${balance.amount} ${feeInfo.denom}.`
+      );
+    }
+  }
+
+  for (let i = 0; i < msgsResponse.msgs.length; i++) {
+    const multiHopMsg = msgsResponse.msgs[i];
+
+    const signerIsLedger = await isLedger(walletClient, multiHopMsg.chain_id);
+
+    let signer: OfflineSigner;
+    if (signerIsLedger) {
+      signer = await getOfflineSignerOnlyAmino(
+        walletClient,
+        multiHopMsg.chain_id
+      );
+    } else {
+      signer = await getOfflineSigner(walletClient, multiHopMsg.chain_id);
+    }
+
+    const chain = getChainByID(multiHopMsg.chain_id);
+
+    const feeInfo = chain.fees?.fee_tokens[0];
+
+    if (!feeInfo) {
+      throw new Error("No fee info found");
+    }
+
+    let gasNeeded = 200000;
+    if (
+      route.does_swap &&
+      route.swap_venue?.chain_id === multiHopMsg.chain_id
+    ) {
+      gasNeeded = 1000000;
+    }
+
+    const msgJSON = JSON.parse(multiHopMsg.msg);
+
+    let msg: EncodeObject;
+
+    let txHash = "";
+
+    if (
+      multiHopMsg.msg_type_url === "/ibc.applications.transfer.v1.MsgTransfer"
+    ) {
+      const client = await getSigningStargateClientForChainID(
+        multiHopMsg.chain_id,
+        signer,
+        {
+          gasPrice: GasPrice.fromString(
+            `${feeInfo.average_gas_price ?? 0}${feeInfo.denom}`
+          ),
+        }
+      );
+
+      msg = {
+        typeUrl: multiHopMsg.msg_type_url,
+        value: {
+          sourcePort: msgJSON.source_port,
+          sourceChannel: msgJSON.source_channel,
+          token: msgJSON.token,
+          sender: msgJSON.sender,
+          receiver: msgJSON.receiver,
+          timeoutHeight: msgJSON.timeout_height,
+          timeoutTimestamp: msgJSON.timeout_timestamp,
+          memo: msgJSON.memo,
+        },
+      };
+
+      if (signerIsLedger) {
+        const currentHeight = await client.getHeight();
+
+        msg.value.timeoutHeight = {
+          revisionHeight: Long.fromNumber(currentHeight).add(100),
+          revisionNumber: Long.fromNumber(currentHeight).add(100),
+        };
+
+        msg.value.timeoutTimestamp = Long.fromNumber(0);
+      }
+
+      if (multiHopMsg.chain_id === "evmos_9001-2") {
+        const tx = await signAndBroadcastEvmos(walletClient, msgJSON.sender, {
```

Comment:
Why do these have custom logic? Do we need to highlight this to users in our docs?